core/keyserver/epochs.go

```diff
@@ -82,18 +63,49 @@ func (s *Server) GetMutations(ctx context.Context, in *pb.GetMutationsRequest) (
 		return nil, status.Error(codes.Internal, "Get signed map root failed")
 	}
 
-	// Get highest and lowest sequence number.
-	meta, err := internal.MetadataFromMapRoot(resp.GetMapRoot())
+	// MapRevisions start at 0. Log leaf indexes starts at 0.
+	// MapRevision should be at least 1 since the Signer is
+	// supposed to create at least one revision on startup.
+	respEpoch := resp.GetMapRoot().GetMapRevision()
+	// Fetch log proofs.
+	logRoot, logConsistency, logInclusion, err := s.logProofs(ctx, domain.LogID, in.GetFirstTreeSize(), respEpoch)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, err
 	}
+	return &pb.Epoch{
+		Smr:            resp.GetMapRoot(),
+		LogRoot:        logRoot.GetSignedLogRoot(),
+		LogConsistency: logConsistency.GetProof().GetHashes(),
+		LogInclusion:   logInclusion.GetProof().GetHashes(),
+	}, nil
+}
 
-	highestSeq := uint64(meta.GetHighestFullyCompletedSeq())
-	lowestSeq, err := s.lowestSequenceNumber(ctx, domain.MapID, in.PageToken, in.Epoch-1)
+// GetEpochStream is a streaming API similar to GetMutations.
+func (s *Server) GetEpochStream(in *pb.GetEpochRequest, stream pb.KeyTransparencyService_GetEpochStreamServer) error {
```

Comment:
you can replace the func args with _ since they're not in use yet.

Reply:
Good idea.

Reply:
Done